src/containerapp/azext_containerapp/_clients.py

```diff
@@ -492,3 +492,106 @@ def list(cls, cmd, resource_group_name, env_name, formatter=lambda x: x):
                 env_list.append(formatted)
 
         return env_list
+
+
+class BuilderClient():
+    api_version = PREVIEW_API_VERSION
+
+    @classmethod
+    def list(cls, cmd, resource_group_name):
+        management_hostname = cmd.cli_ctx.cloud.endpoints.resource_manager
+        sub_id = get_subscription_id(cmd.cli_ctx)
+        url_fmt = "{}/subscriptions/{}/resourceGroups/{}/providers/Microsoft.App/builders?api-version={}"
+        request_url = url_fmt.format(
+            management_hostname.strip('/'),
+            sub_id,
+            resource_group_name,
+            cls.api_version)
+        r = send_raw_request(cmd.cli_ctx, "GET", request_url)
+        return r.json()
+
+    @classmethod
+    def create(cls, cmd, builder_name, resource_group_name, environment_name, location, no_wait=False):
+        management_hostname = cmd.cli_ctx.cloud.endpoints.resource_manager
+        sub_id = get_subscription_id(cmd.cli_ctx)
+        url_fmt = "{}/subscriptions/{}/resourceGroups/{}/providers/Microsoft.App/builders/{}?api-version={}"
+        request_url = url_fmt.format(
+            management_hostname.strip('/'),
+            sub_id,
+            resource_group_name,
+            builder_name,
+            cls.api_version)
+        body_data = {
+            "location": location,
+            "properties": {
+                "environmentId": f"/subscriptions/{sub_id}/resourcegroups/{resource_group_name}/providers/microsoft.app/managedenvironments/{environment_name}"
```

Comment:
could you use same pattern with other resource id?
resourcegroup -> resourceGroup
microsoft.app -> Microsoft.App 
...

Reply:
Updated! Thanks